golangci-lint/.golangci.yml

```diff
@@ -30,73 +30,112 @@ linters-settings:
           - (github.com/golangci/golangci-lint/pkg/logutils.Log).Errorf
           - (github.com/golangci/golangci-lint/pkg/logutils.Log).Fatalf
-  maligned:
-    suggest-new: true
   misspell:
     locale: US
   nolintlint:
     allow-leading-space: true # don't require machine-readable nolint directives (i.e. with no leading space)
     allow-unused: false # report any unused nolint directives
     require-explanation: false # don't require an explanation for nolint directives
-    require-specific: false # don't require nolint directives to be specific about which linter is being skipped
+    require-specific: true # require nolint directives to be specific about which linter is being skipped
 
 linters:
   # please, do not use `enable-all`: it's deprecated and will be removed soon.
   # inverted configuration with `enable-all` and `disable` is not scalable during updates of golangci-lint
   disable-all: true
   enable:
+    - deadcode
+    - errcheck
+    - gosimple
+    - govet
+    - ineffassign
+    - staticcheck
+    - structcheck
+    - typecheck
+    - unused
+    - varcheck
+    # Disabled by default linters
+    - asciicheck
+    - bidichk
     - bodyclose
+    - containedctx
+    - contextcheck
+    - cyclop
     - deadcode
     - depguard
     - dogsled
     - dupl
+    - durationcheck
     - errcheck
-    - exportloopref
+    - errname
+    - errorlint
+    - execinquery
     - exhaustive
+    # Initializing structs with all fields is nonsense.
+    # - exhaustruct
+    - exportloopref
+    - forbidigo
+    - forcetypeassert
     - funlen
+    - gci
+    - gochecknoglobals
     - gochecknoinits
+    - gocognit
+    - goconst
     - gocritic
     - gocyclo
-    - gocognit
-    - goerr113    #only add if go > 1.13
-    - errorlint   #only add if go > 1.13
+    - godot
+    # TODOs are fine.
+    #- godox
+    - goerr113
     - gofmt
+    - gofumpt
+    - goheader
     - goimports
-    - golint
     - gomnd
+    - gomoddirectives
+    - gomodguard
     - goprintffuncname
     - gosec
-    - gosimple
-    - govet
-    - ineffassign
-    - maligned
+    - grouper
+    - ifshort
+    - imrtrasr
+    - ireturn
+    # lll is flagged pretty much on tests only.
+    #- lll
+    - maintidx
+    - makezero
     - misspell
+    - nakedtjson
     - nestif
     - nilerr
+    - nilnil
+    # NLReturn feels like a white label wsl, more annoying for what it's worth.
+    #- nlreturn
     - noctx
+    - nolintlint
+    - nonamedreturns
+    - nosprintfhostport
+    - paralleltest
     - prealloc
+    - predeclared
+    - promlinter
+    - revive
     - rowserrcheck
-    - staticcheck
-    - structcheck
+    - sqlclosecheck
     - stylecheck
-    - typecheck
+    # Tags formats vary depending on the project so tagliatelle des not help in common configuration.
+    #- tagliatelle
+    - tenv
+    # _test packages help to test the APIs but internal tests can also be useful, so we do not enforce the black-box approach for all tests.
+    # - testpackage
```

Comment:
😢  i actually like this one

Reply:
I like doing pkg `_test` tests but I don't think all tests should be like that. 

From [the linter description](https://github.com/maratori/testpackage):
> The linter reports if a test is in a package without suffix _test. If you really need to test unexported function, then put the test into file XXX_internal_test.go. The linter skips such files by default. It also skips the file export_test.go by default (see the last article below).

I think testing internal things can be actually useful and doesn't have to be an exception...